mitxgraders/helpers/calc/expressions.py

```diff
@@ -381,57 +267,69 @@ def cast_np_numeric_as_builtin(obj, map_across_lists=False):
                 for item in obj]
     return obj
 
-class FormulaParser(object):
+class MathParser(object):
     """
-    Parses a mathematical expression into a tree that can subsequently be evaluated
-    against given dictionaries of variables and functions.
+    Parses mathematical expressions into trees and caches the result.
+    Expression trees are returned as MathExpression objects, which can then
+    be evaluated.
+
+    Usage
+    =====
+    >>> new_parser = MathParser()
+    >>> parsed = new_parser.parse('2*x + 5')
+    >>> isinstance(parsed, MathExpression)
+    True
+    >>> parsed
+    <BLANKLINE>
```

Comment:
This is curious? What's with the <BLANKLINE>?

Reply:
It's a blank line :).

I have no idea what's causing that. It happens in master branch, too. Atom and iTerm2 print it as an actual blank line, except when run in a doctest. Doctest shows it as a <BLANKLINE>. 

Anyway, it never reared its head before because we didn't have a test for FormulaParser.dump_parse_result (it had `pragma: no-cover`). But I wrote a test for the printing, so now we see it.

But I don't know where it comes from ... Maybe built in to pyparsing's `asxML()` method?